spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -747,69 +814,51 @@ function cancelChannelUpgrade(
+        abortTransactionUnless(verifyChannelUpgradeError(connection, proofHeight, proofUpgradeError, currentChannel.counterpartyPortIdentifier, currentChannel.counterpartyChannelIdentifier, errorReceipt))
+    }
 
-    // call modules onChanUpgradeRestore callback
-    module = lookupModule(portIdentifier)
-    // restore callback must not return error since counterparty successfully restored previous channelEnd
-    module.onChanUpgradeRestore(
-        portIdentifer,
-        channelIdentifier
-    )
+    // cancel upgrade and write error receipt
+    restoreChannel(portIdentifier, channelIdentifier)
 }
 ```
 
 ### Timeout Upgrade Process
 
-It is possible for the channel upgrade process to stall indefinitely on TRYUPGRADE if the TRYUPGRADE transaction simply cannot pass on the counterparty; for example, the upgrade feature may not be enabled on the counterparty chain.
-
-In this case, we do not want the initializing chain to be stuck indefinitely in the `INITUPGRADE` step. Thus, the `ChannelUpgradeInitMsg` message will contain a `TimeoutHeight` and `TimeoutTimestamp`. The counterparty chain is expected to reject `ChannelUpgradeTryMsg` message if the specified timeout has already elapsed.
-
-A relayer must then submit an `ChannelUpgradeTimeoutMsg` message to the initializing chain which proves that the counterparty is still in its original state. If the proof succeeds, then the initializing chain shall also restore its original channel to `OPEN` and cancel the upgrade.
+It is possible for the channel upgrade process to stall indefinitely while trying to flush the existing packets. To protect against this, each chain sets a timeout when it moves into `FLUSHING`. If the counterparty has not completed the handshake within the expected time window, then the relayer can submit a timeout message to restore the channel to OPEN with the original parameters. It will also write an error receipt so that the counterparty which has not moved to `FLUSHCOMPLETE` can also restore channel to OPEN with the original parameters.
 
 ```typescript
 function timeoutChannelUpgrade(
     portIdentifier: Identifier,
     channelIdentifier: Identifier,
     counterpartyChannel: ChannelEnd,
-    prevErrorReceipt: ErrorReceipt, // optional
     proofChannel: CommitmentProof,
-    proofErrorReceipt: CommitmentProof,
     proofHeight: Height,
 ) {
-    // current channel must be in INITUPGRADE
+    // current channel must have an upgrade that is FLUSHING or FLUSHCOMPLETE
+    upgrade = provableStore.get(channelUpgradePath(portIdentifier, channelIdentifier))
+    abortTransactionUnless(upgrade != nil)
     currentChannel = provableStore.get(channelPath(portIdentifier, channelIdentifier))
-    abortTransactionUnles(currentChannel.state == INITUPGRADE)
+    abortTransactionUnless(currentChannel.flushStatus != FLUSHING || currentChannel.flushStatus == FLUSHCOMPLETE)
```

Comment:
According to the comment `// current channel must have an upgrade that is FLUSHING or FLUSHCOMPLETE` above...

```suggestion
    abortTransactionUnless(currentChannel.flushStatus != FLUSHING || currentChannel.flushStatus != FLUSHCOMPLETE)
```